modules/api/src/main/java/org/eclipse/xpanse/api/controllers/CspServiceTemplateApi.java

```diff
@@ -117,42 +120,67 @@ public List<ServiceTemplateDetailVo> listManagedServiceTemplates(
     /**
      * View details of service template registration.
      *
-     * @param id id of service template.
+     * @param serviceTemplateId service template id.
      * @return service template details.
      */
     @Tag(name = "CloudServiceProvider",
             description = "APIs for cloud service provider to manage service templates.")
     @Operation(description = "view service template by id.")
-    @GetMapping(value = "/csp/service_templates/{id}", produces =
+    @GetMapping(value = "/csp/service_templates/{serviceTemplateId}", produces =
             MediaType.APPLICATION_JSON_VALUE)
     @ResponseStatus(HttpStatus.OK)
     @AuditApiRequest(methodName = "getCspFromServiceTemplateId")
-    public ServiceTemplateDetailVo getRegistrationDetails(
-            @Parameter(name = "id", description = "id of service template")
-            @PathVariable("id") String id) {
+    public ServiceTemplateDetailVo getServiceTemplateDetails(
+            @Parameter(name = "serviceTemplateId", description = "id of service template")
+            @PathVariable("serviceTemplateId") UUID serviceTemplateId) {
         ServiceTemplateEntity templateEntity =
-                serviceTemplateManage.getServiceTemplateDetails(UUID.fromString(id), false, true);
+                serviceTemplateManage.getServiceTemplateDetails(serviceTemplateId, false, true);
         return convertToServiceTemplateDetailVo(templateEntity);
     }
 
+
     /**
-     * Review service template registration.
+     * List pending service template request history to review.
      *
-     * @param id                        id of service template.
-     * @param reviewRegistrationRequest review request for service template registration.
+     * @return service templates
      */
     @Tag(name = "CloudServiceProvider",
             description = "APIs for cloud service provider to manage service templates.")
-    @Operation(description = "Review service template registration.")
-    @PutMapping(value = "/service_templates/review/{id}", produces =
+    @Operation(description = "List service template requests history pending to review.")
```

Comment:
Get service template requests pending to review.

Reply:
updated.